pulsar-client/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -46,7 +46,7 @@
     private boolean blockIfQueueFull = false;
     private int maxPendingMessages = 1000;
     private MessageRoutingMode messageRouteMode = MessageRoutingMode.SinglePartition;
-    private boolean useMurmurHash = true;
+    private HashingScheme hashingScheme = HashingScheme.Murmur3_32Hash;
```

Comment:
We cannot change the default from Java to Murmur hash, since it would unexpectedly break ordering for people upgrading the library.

Reply:
@merlimat Is this also can be said to C++ client?

Reply:
Yes, unfortunately we cannot change the default in either library implementation. C++ would have to keep using the current BoostHash